src/stores/analyticsContext.test.tsx

```diff
@@ -44,13 +43,18 @@ window.fetch = jest.fn(() =>
 
 const mockedFetch = fetch as jest.Mock
 
+// ignore console.debug and don't output to test output
+jest.spyOn(console, 'debug').mockImplementation(jest.fn())
+const mockConsoleWarn = jest.fn()
+jest.spyOn(console, 'warn').mockImplementation(mockConsoleWarn)
+
 describe('useAnalytics', () => {
-  it('throws an error if AnalyticsContext is undefined', () => {
+  test('throws an error if AnalyticsContext is undefined', () => {
     jest.spyOn(React, 'useContext').mockReturnValueOnce(undefined)
     expect(() => useAnalytics()).toThrowError()
```

Comment:
I suspect this test is not testing what it is expecting. Also running this suite results in console log below that leads me to believe the tests here need to be refactored. But this PR is already large enough so I left it for now. Would love thoughts on this.

```sh
  console.error
    Warning: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
    1. You might have mismatching versions of React and the renderer (such as React DOM)
    2. You might be breaking the Rules of Hooks
    3. You might have more than one copy of React in the same app
    See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.

      158 |
      159 | export const useAnalytics = () => {
    > 160 |   const context = useContext(AnalyticsContext)
          |                             ^
      161 |
      162 |   if (context === undefined) {
      163 |     throw new Error(

      at printWarning (node_modules/react/cjs/react.development.js:209:30)
      at error (node_modules/react/cjs/react.development.js:183:7)
      at resolveDispatcher (node_modules/react/cjs/react.development.js:1592:7)
      at useContext (node_modules/react/cjs/react.development.js:1602:20)
      at useAnalytics (src/stores/analyticsContext.tsx:160:29)
      at src/stores/analyticsContext.test.tsx:54:30
      at Object.<anonymous> (node_modules/expect/build/toThrowMatchers.js:74:11)
      at Object.throwingMatcher [as toThrowError] (node_modules/expect/build/index.js:312:21)
      at Object.toThrowError (src/stores/analyticsContext.test.tsx:54:34)

 PASS  src/stores/analyticsContext.test.tsx
  useAnalytics
    ✓ throws an error if AnalyticsContext is undefined (28 ms)
    ✓ returns the created context (6 ms)
  Analytics context
    ✓ warns in the console if missing the analytics URL (55 ms)
    ✓ warns in the console if missing the analytics site ID (54 ms)
    ✓ initializes the analytics queue on render (5 ms)
    with config defined
      ✓ initializes the analytics config on mount (4 ms)
      ✓ provides the push handler (41 ms)
      ✓ provides the trackEvent handler (15 ms)
      ✓ can use trackEvent to track an event name (15 ms)
      ✓ can use trackEvent to track an event value (14 ms)

```

Reply:
my understanding is the console error is what we expect from that test -- if the hook is used improperly, it'll throw that error. we have a similar function/test in the other contexts as well. maybe i'm missing something?

Reply:
Yeah I agree with @abbyoung that the console error looks to be expected behavior, but the error message in `useAnalytics` is 'useAnalytics must be used within an AnalyticsContextProvider' and I'm not seeing that anywhere in the output. So maybe that is an issue? Maybe we should be seeing that specific error message as opposed to the bulleted list in the above read out.

Reply:
iirc I tried to make the test be more specific by looking at what error was returned and had trouble getting it to work. I was hoping to avoid printing expected errors to the logs in our tests since it's frequently a source of confusion for me. But I think this was one I couldn't fix. I'm leaving it as is for now :)